python/pyspark/mllib/linalg/__init__.py

```diff
@@ -122,6 +123,15 @@ def _format_float_list(l):
     return [_format_float(x) for x in l]
 
 
+def _double_to_long_bits(value):
```

Comment:
We can make the code more readable:

~~~python
if isnan(value):
  value = float('nan')
return struct.unpack('Q', struct.pack('d', value))[0]
~~~